lib/workers/repository/update/branch/index.ts

```diff
@@ -79,8 +80,22 @@ export interface ProcessBranchResult {
   result: BranchResult;
 }
 
+function canSkipBranchUpdateCheck(
+  branchName: string,
+  branchCache: BranchCache,
+  branchFingerprint: string
+): boolean {
+  const branchCommitSha = getBranchCommit(branchName);
+
+  return (
+    branchCommitSha === branchCache.sha &&
+    branchFingerprint === branchCache.branchFingerprint
+  );
```

Comment:
I would like to move the skip update logging to this function and log both cases:
- When an existing fingerprint exists but does not match
- When there's a match

By logging both these cases (and not logging when there is no fingerprint) then we can measure the % of cache success